python/kotekan/runner.py

```diff
@@ -136,9 +138,17 @@ def run(self):
             # Print out the output from Kotekan for debugging
             print(self.output)
 
-            # Throw an exception if we don't exit cleanly
-            if p.returncode:
-                raise subprocess.CalledProcessError(p.returncode, cmd)
+            # If failure is expected just report the exit code
+            if self.expect_failure is True:
+                print("Test failed as expected with exit code: " + p.returncode)
+                self.return_code = p.returncode
+            else:
+                try:
```

Comment:
This block is a bit odd in that it raises an exception which it immediately catches and then doesn't propagate (which it should). I think you can replace the whole `else` branch with:

```
elif p.returncode:
    raise subprocess.CalledProcessError(p.returncode, cmd)
```